### PR TITLE
feat(unlock-js): add Unlock v13 and PublicLock v14 support

### PR DESCRIPTION
**Pull request by clemsos**
# Description

This adds support Unlock v13 and PublicLock v14 to unlock-js

# Issues

Fixes #
Refs #

# Checklist:

- [ ] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: packages/unlock-js/scripts/compressAbi.js [hunk 1]
```diff
@@ -11,7 +11,17 @@ const fs = require('fs')
 const path = require('path')
 const abis = require('@unlock-protocol/contracts')
 
-const unlockVersions = ['v4', 'v6', 'v7', 'v8', 'v9', 'v10', 'v11', 'v12']
+const unlockVersions = [
+  'v4',
+  'v6',
+  'v7',
+  'v8',
```
Comment (julien51) [inline packages/unlock-js/scripts/compressAbi.js:18]: I think we should consider cleaning up these older versions... wdyt?

File: packages/unlock-js/scripts/compressAbi.js [hunk 2]
```diff
@@ -11,7 +11,17 @@ const fs = require('fs')
...
+  'v9',
+  'v10',
+  'v11',
+  'v12',
+  'v13',
+]
 const publicLockVersions = [
```
Comment (julien51) [inline packages/unlock-js/scripts/compressAbi.js:25]: same here. I'd sayd anything that has not seen any activity in 2 years can be removed

File: packages/unlock-js/src/PublicLock/v14/index.js [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import abis from '../../abis'
+import v13 from '../v13'
+
+export default {
+  ...v13,
+  version: 'v14',
+  PublicLock: abis.PublicLock.v14,
```
Comment (julien51) [inline packages/unlock-js/src/PublicLock/v14/index.js:7]: No function change?

**Review by julien51 — APPROVED** [review]

File: packages/unlock-js/src/PublicLock/v14/index.js (hunk 3 above)
Comment (clemsos) [reply packages/unlock-js/src/PublicLock/v14/index.js:7]: we added a `setExpirationKey` but I wasn't sure we'll need it in unlockjs rn

File: packages/unlock-js/scripts/compressAbi.js (hunk 2 above)
Comment (clemsos) [reply packages/unlock-js/scripts/compressAbi.js:25]: yes will do in another PR

**Pull request merged**